multi-node/aws/pkg/config/templates/cloud-config-controller

```diff
@@ -91,17 +93,18 @@ coreos:
       content: |
         [Unit]
         Description=decrypt kubelet tls assets using amazon kms
-        Before=kubelet.service
-        After=docker.service
-        Requires=docker.service
+        Before=kubelet.service flanneld.service
```

Comment:
Dependency injection is a distinct pattern, the inverse of traditional `Requires` dependency listing. Desiring to standardize on one or the other is like requiring a road-system only have right-hand turns for simplicity sake (cough new jersey).

A textbook use-case for dependency injection (`RequiredBy`) is pluggable dependencies (aka providers). The kubelet needs a "network provider", that can be flannel, weave, AWS sdk, etc. Defining and enabling the `flanneld.service` will transparently inject `flanneld.service` as a dependency of the kubelet, which is much preferable to mutating `kubelet.service` every time you want to change network providers.